lib/services/configuration-files.js

```diff
@@ -0,0 +1,113 @@
+import { VERSIONED_CONFIGURATION_WATCHER, VERSIONED_CONFIGURATION_ROOT } from '../../env';
+import moment from 'moment';
+import { SemanticFile } from '../entities/semantic-file';
+import { DirectorySynchronizer } from './directory-synchronizer';
+import { Gaze } from 'gaze';
+import { sleep } from '../util/generic';
+import { SemanticFormSpecification } from '../entities/semantic-form-specification';
+
+/**
+ * Service providing the up-to-date and synced-up configuration-files.
+ */
+export class ConfigurationFiles {
+
+  constructor() {
+    this.directory = new DirectorySynchronizer(VERSIONED_CONFIGURATION_ROOT);
+    this.debug = true;
+    this.watcher = VERSIONED_CONFIGURATION_WATCHER;
+  }
+
+  /**
+   * Initialization of the service.
+   */
+  async init() {
+    await this.directory.sync();
+    await this.sync();
+    return this;
+  }
+
+  /**
+   * Returns the latest specification.
+   *
+   * @returns {SemanticFormSpecification}
+   */
+  get specification() {
+    return new SemanticFormSpecification(this.directory.getLatest());
+  }
+
+  /**
+   * Returns the mapper configuration file.
+   *
+   * @returns {undefined|SemanticFile}
+   */
+  get mapper() {
+    const file = new SemanticFile({filename: '/config/mapper.js'});
+    if (SemanticFile.exists(file)) {
+      return file;
+    }
+    throw '[ERROR] Missing configuration file: mapper.js';
+  }
+
+  /**
+   * Setup a watcher to sync-up configuration-files when new files are added to the root.
+   *
+   * TODO:  Triggered for EVERY file change while our service re-syncs all files @ a time.
+   *        Therefore this is sub-optimal, as it triggers a full refresh for every file.
+   *
+   * @param enabled
+   */
+  set watcher(enabled) {
+    this.gaze = new Gaze(`${VERSIONED_CONFIGURATION_ROOT}/**`);
+    if (enabled) {
+      console.log('Gazing over the configuration-files o.0');
+      this.gaze.on('all', async () => {
+        console.log(
+            `Configuration-files sync. triggered by watcher at ${moment()}, waiting 1 sec. before starting ...`);
+        await sleep(1000);
+        try {
+          await this.sync();
+        } catch (e) {
+          console.error(`Configuration-files sync. watcher FAILED at${moment()}`);
+          console.error(e);
+        }
+      });
+    } else {
+      this.gaze.close();
```

Comment:
Couldn't we just create the Gaze if `enabled=true` instead of creating and then closing it ?